utam-core/src/main/java/utam/core/element/Element.java

```diff
@@ -19,56 +20,154 @@
+   *
+   * @param by                 locator
+   * @param isExpandShadowRoot if true, search inside element's shadowRoot
+   * @return number of elements or 0 if none found
+   */
   int containsElements(Locator by, boolean isExpandShadowRoot);
 
+  /**
+   * check if element is "displayed" according to underlying implementing library
+   *
+   * @return boolean, true if element is displayed
+   */
   boolean isDisplayed();
 
+  /**
+   * check if element is "enabled" according to underlying implementing library
+   *
+   * @return boolean, true if element is enabled
+   */
   boolean isEnabled();
 
+  /**
+   * check if element is present on the page. It returns false if it's a null or became stale
+   *
+   * @return boolean
+   */
   boolean isExisting();
 
+  /**
+   * clear text inside the element, usually for an input box
+   */
   void clear();
 
+  /**
+   * click on the element
+   */
   void click();
 
   /**
    * provided as temporary workaround when regular click does not work
-   * @param driver Driver instance used in deprecated click
+   *
    * @deprecated when all browsers work as expected, will be removed
    */
   @Deprecated
-  void deprecatedClick(Driver driver);
+  void deprecatedClick();
 
+  /**
+   * get value of the attribute
+   *
+   * @param attrName name of the attribute
+   * @return returned value
+   */
   String getAttribute(String attrName);
 
+  /**
+   * get text inside an element
+   *
+   * @return string
+   */
   String getText();
 
+  /**
+   * type in text inside the element, does not clear it first
+   *
+   * @param text string to enter
+   */
   void setText(String text);
 
-  void scrollIntoView(Driver driver, ScrollOptions options);
+  /**
+   * scroll the element into view
+   *
+   * @param options scroll into view options
+   */
+  void scrollIntoView(ScrollOptions options);
 
-  void moveTo(Driver driver);
+  /**
+   * move cursor to the element
+   */
+  void moveTo();
 
-  boolean hasFocus(Driver driver);
+  /**
+   * check if element has focus
+   *
+   * @return boolean
+   */
+  boolean hasFocus();
 
-  void blur(Driver driver);
+  /**
+   * blur the element
+   */
+  void blur();
 
-  void focus(Driver driver);
+  /**
+   * focus on the element
+   */
+  void focus();
 
-  void flick(Driver driver, int xOffset, int yOffset);
+  /**
+   * only for mobile elements - flick action based on offset coordinates
+   *
+   * @param xOffset horizontal offset
+   * @param yOffset vertical offset
+   */
+  void flick(int xOffset, int yOffset);
 
+  /**
+   * only for mobile elements - flick action based on direction
+   *
+   * @param direction direction of the flick
+   * @return boolean
+   */
   boolean flickItems(GestureDirection direction);
 
+  /**
+   * drag and drop an element to the target location
+   *
+   * @param options location where to drop as element or coordinates offset and optional hold
+   *                duration
+   */
+  void dragAndDrop(DragAndDropOptions options);
```

Comment:
@jimevanssfdc pls check if you're ok with API for dragAndDrop, mainly passing options as a single type, DragAndDropOptions is declared below